### PR TITLE
Issue #410 Allow OpenShift version to be specified without 'v' prefix

### PR DESCRIPTION
**Pull request by hferentschik**
- Added more concise error message for the case that unparsable version gets specified
- Updated tests

Addresses #410

### CODE REVIEW

**Comment by hferentschik** [comment]
Based on top of pull request #962 which should be merged first.

**Comment by hferentschik** [comment]
Rebased on latest #962

File: cmd/minishift/cmd/start.go [hunk 1]
```diff
@@ -87,6 +82,13 @@ var minishiftToClusterUp = map[string]string{
 
 // init configures the command line options of this command
 func init() {
+	startCmd = &cobra.Command{
+		Use:   commandName,
+		Short: "Starts a local OpenShift cluster.",
+		Long:  `Starts a local single-node OpenShift cluster on the specified hypervisor.`,
```
Comment (budhrg) [inline cmd/minishift/cmd/start.go]: Hyphen is not require in `single-node` phrase.

Comment (budhrg) [inline cmd/minishift/cmd/start.go]: `on the specified hypervisor` -> `using the native hypervisor`. I think we specify when we need.

File: pkg/minishift/clusterup/clusterup.go [hunk 2]
```diff
@@ -121,9 +121,10 @@ func EnsureHostDirectoriesExist(host *host.Host, dirs []string) error {
 // DetermineOpenShiftVersion returns the OpenShift/oc version to use.
 // If the requested version is < the base line version we use baseline oc to provision the requested OpenShift version.
 // If the requested OpenShift version is >= the baseline, we align the oc version with the requested OpenShift version.
-// It also handles the case the user just specifies a version without the required 'v' prefix
 func DetermineOpenShiftVersion(requestedVersion string) string {
-	if !ValidateOpenshiftMinVersion(requestedVersion, version.GetOpenShiftVersion()) {
+	valid, _ := ValidateOpenshiftMinVersion(requestedVersion, version.GetOpenShiftVersion())
+
```
Comment (budhrg) [inline pkg/minishift/clusterup/clusterup.go]: unnecessary extra line at line 126.

File: pkg/minishift/clusterup/clusterup_test.go [hunk 3]
```diff
@@ -16,24 +16,44 @@ limitations under the License.
...
-		"v1.3.5-alpha":   false,
-		"v1.4.1":         true,
-		"v1.5.0-alpha.0": true,
-		"v1.5.1-beta.0":  true,
-		"v1.6.0":         true,
+	var versionTests = []struct {
+		version string // input
+		valid   bool   // expected result
+		err     error
+	}{
```
Comment (budhrg) [inline pkg/minishift/clusterup/clusterup_test.go]: :+1:

**Comment by budhrg** [comment]
Functionally working for me: 
```
$ minishift start --openshift-version 1.5.0
Starting local OpenShift cluster using 'kvm' hypervisor...
Downloading OpenShift binary 'oc' version 'v1.5.1'
 19.96 MiB / 19.96 MiB [==========================================================================================================================] 100.00% 0s
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... OK
-- Checking for existing OpenShift container ... OK
-- Checking for openshift/origin:v1.5.0 image ... 
   Pulling image openshift/origin:v1.5.0
.....
```

One extra line found and suggested full description of command, rest seems to good to me :+1:

File: cmd/minishift/cmd/start.go (hunk 1 above)
Comment (hferentschik) [inline cmd/minishift/cmd/start.go]: Nothing I originally changed, but I agree that "on the specified hypervisor" seems odd. To be honest I am not sure why we single out the hypervisor. `start` has a whole bunch of important parameters. I think I prefer to remove this clause altogether.

File: pkg/minishift/clusterup/clusterup.go (hunk 2 above)
Comment (hferentschik) [inline pkg/minishift/clusterup/clusterup.go]: Ok

**Comment by hferentschik** [comment]
pushed an update...

File: cmd/minishift/cmd/start.go [hunk 4]
```diff
@@ -87,6 +82,13 @@ var minishiftToClusterUp = map[string]string{
 
 // init configures the command line options of this command
 func init() {
+	startCmd = &cobra.Command{
```
Comment (gbraad) [inline cmd/minishift/cmd/start.go]: Why breaking consistency with previous implementations? Non of the commands have been defined like this inside the `init` function...

Comment (budhrg) [inline cmd/minishift/cmd/start.go]: +1

Comment (hferentschik) [inline cmd/minishift/cmd/start.go]: You have to, otherwise you are getting a compilation error

Comment (LalatenduMohanty) [inline cmd/minishift/cmd/start.go]: @hferentschik Right, wondering why is this so. `runStart` anyways runs after `init()`.

Comment (gbraad) [inline cmd/minishift/cmd/start.go]: you mean, this is due to the way the flags are used in the validate function?

**Comment by LalatenduMohanty** [comment]
I have tested it and works for me.

File: cmd/minishift/cmd/start.go (hunk 4 above)
Comment (gbraad) [inline cmd/minishift/cmd/start.go]: Ah, I see... cyclic dependency between the functions... :-s

Comment (LalatenduMohanty) [inline cmd/minishift/cmd/start.go]: I think below error shows that in getting to initialization loop which has cyclic dependency 
```
$ make
go install -pkgdir=/home/lmohanty/GoCode/src/github.com/minishift/minishift/out/bindata -ldflags="-X github.com/minishift/minishift/pkg/version.version=1.0.1 -X github.com/minishift/minishift/pkg/version.isoVersion=v1.0.2 -X github.com/minishift/minishift/pkg/version.openshiftVersion=v1.5.1" ./cmd/minishift
# github.com/minishift/minishift/cmd/minishift/cmd
cmd/minishift/cmd/start.go:70: initialization loop:
	/home/lmohanty/GoCode/src/github.com/minishift/minishift/cmd/minishift/cmd/start.go:70 startCmd refers to
	/home/lmohanty/GoCode/src/github.com/minishift/minishift/cmd/minishift/cmd/start.go:104 runStart refers to
	/home/lmohanty/GoCode/src/github.com/minishift/minishift/cmd/minishift/cmd/start.go:384 validateOpenshiftVersion refers to
	/home/lmohanty/GoCode/src/github.com/minishift/minishift/cmd/minishift/cmd/start.go:70 startCmd
Makefile:56: recipe for target '/home/lmohanty/GoCode/bin/minishift' failed
make: *** [/home/lmohanty/GoCode/bin/minishift] Error 2
```

Comment (gbraad) [inline cmd/minishift/cmd/start.go]: cyclic dependency...

Comment (LalatenduMohanty) [inline cmd/minishift/cmd/start.go]: @hferentschik Lets add a comment  why startCmd moved to init() and then we are good to merge. cc @gbraad

**Comment by hferentschik** [comment]
@LalatenduMohanty added comment to init in start.go. Please go ahead and merge.

**Pull request merged**